docs/api/locking.md

```diff
@@ -0,0 +1,449 @@
+# Git LFS File Locking API
+
+Added: v2.0
+
+The File Locking API is used to create, list, and delete locks, as well as
+verify that locks are respected in Git pushes. The locking URLs are built
+by adding a suffix to the LFS Server URL.
+
+Git remote: https://git-server.com/foo/bar  
+LFS server: https://git-server.com/foo/bar.git/info/lfs  
+Locks API: https://git-server.com/foo/bar.git/info/lfs/locks
+
+See the [Server Discovery doc](./server-discovery.md) for more info on how LFS
+builds the LFS server URL.
+
+All File Locking requests require the following HTTP headers:
+
+    Accept: application/vnd.git-lfs+json
+    Content-Type: application/vnd.git-lfs+json
+
+See the [Authentication doc](./authentication.md) for more info on how LFS
+gets authorizes Batch API requests.
+
+Note: This is the first version of the File Locking API, supporting only the
+simplest use case: single branch locking. The API is designed to be extensible
+as we experiment with more advanced locking scenarios, as defined in the
+[original proposal](/docs/proposals/locking.md).
+
+## Create Lock
+
+The client sends the following to create a lock by sending a `POST` to `/locks`
+(appended to the LFS server url, as described above).
+
+* `path` - String path name of the file that is locked. This should be
+relative to the root of the repository working directory.
+* `committer` - The optional values of `user.name` and `user.email` from the
+git config.
+
+Lock services are not required to use or store the committer details, since most
+lock services should already be able to identify the committer according to
+their credentials. The `committer` values are used for display purposes only.
+
+```js
+// POST https://lfs-server.com/locks
+// Accept: application/vnd.git-lfs+json
+// Content-Type: application/vnd.git-lfs+json
+// Authorization: Basic ...
+{
+  "path": "foo/bar.zip",
+  "committer": {
+    "name": "Jane Doe",
+    "email": "janedoe@email.com"
+  }
+}
+```
+
+### Successful Response
+
+Successful responses return the created lock:
+
+* `id` - String ID of the Lock. Git LFS doesn't enforce what type of ID is used,
+as long as it's returned a string.
+* `path` - String path name of the locked file. This should be relative to the
+root of the repository working directory.
+* `locked_at` - The string ISO 8601 formatted timestamp the lock was created.
+* `committer` - The optional name/email of the user that created the Lock. This
```

Comment:
As a server response I think we should either require the server to provide the user information or remove it from the API. Leaving it optional means it's essentially useless to us because we can't rely on it.